src/main/java/org/mule/extensions/java/internal/operation/JavaInvokeOperations.java

```diff
@@ -0,0 +1,124 @@
+import org.mule.runtime.extension.api.annotation.metadata.MetadataKeyId;
+import org.mule.runtime.extension.api.annotation.metadata.OutputResolver;
+import org.mule.runtime.extension.api.annotation.metadata.TypeResolver;
+import org.mule.runtime.extension.api.annotation.param.Content;
+import org.mule.runtime.extension.api.annotation.param.NullSafe;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+import org.mule.runtime.extension.api.annotation.param.ParameterGroup;
+
+import java.lang.reflect.Method;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+/**
+ * Defines the operations of {@link JavaModule} related to invocation of class or instance methods using reflection.
+ *
+ * @since 1.0
+ */
+public class JavaInvokeOperations {
+
+  @Inject
+  private JavaModuleLoadingCache cache;
+
+  /**
+   * Operation that allows the user to invoke static methods with the provided arguments.
+   * The identifier of the {@link Method} to be invoked includes the {@code class} and {@code method} names,
+   * being the {@code method} a full description of its signature including the types of each parameter.
+   * <p>
+   * For example, if we want to invoke the static method {@code echo} with signature {@code public static String echo(String msg)}
+   * which belongs to {@link Class} {@code org.bar.Foo}, then the identifier of the method will be {@code "echo(String)"}
+   *
+   * @param identifier the unique identifier for the method to be invoked.
+   * @param args       the arguments used to invoke the given {@link Method}
+   * @return the result of the {@link Method} invocation with the given {@code args}
+   * @throws ClassNotFoundModuleException    if the given {@code class} is not found in the current context
+   * @throws NoSuchMethodModuleException     if the given {@code class} does not declare a method with the given signature
+   * @throws ArgumentMismatchModuleException if the {@code method} requires a different set of arguments than the ones provided
+   * @throws InvocationModuleException       if an error occurs during the execution of the method
+   */
+  @Throws(JavaInvokeErrorProvider.class)
+  @OutputResolver(output = StaticMethodTypeResolver.class)
+  public Object invokeStatic(
+                             @ParameterGroup(
+                                 name = "Method") @MetadataKeyId(StaticMethodTypeResolver.class) StaticMethodIdentifier identifier,
+                             @Optional @NullSafe @Content @TypeResolver(StaticMethodTypeResolver.class) Map<String, TypedValue<Object>> args)
+      throws ClassNotFoundModuleException, ArgumentMismatchModuleException,
+      InvocationModuleException, NoSuchMethodModuleException {
+
+    Class<?> targetClass = cache.loadClass(identifier.getClazz())
+        .orElseThrow(
+                     () -> new ClassNotFoundModuleException(format("Failed to load Class with name [%s] while invoking static Method [%s]",
```

Comment:
no need to say in the error "while invoking static method"... Mule automatically logs the processor that failed alongside its parameter